src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptingService.cs

```diff
@@ -31,52 +83,168 @@ public void InitializeService(ServiceHost serviceHost)
         /// </summary>
-        /// <param name="scriptingParams"></param>
-        /// <param name="requestContext"></param>
-        internal static async Task HandleScriptingScriptAsRequest(
-            ScriptingScriptAsParams scriptingParams,
-            RequestContext<ScriptingScriptAsResult> requestContext)
+        private static string ScriptAsCreate(
+            IBindingContext bindingContext,
+            ConnectionInfo connInfo,
+            ObjectMetadata metadata)
         {
-            string script = string.Empty;
-            if (scriptingParams.Operation == ScriptOperation.Select)
+            Scripter scripter = new Scripter(bindingContext.ServerConnection, connInfo);
+            StringCollection results = null;
+            if (metadata.MetadataType == MetadataType.Table)
             {
-                script = string.Format(
-@"SELECT *
-FROM {0}.{1}",
-                scriptingParams.Metadata.Schema, scriptingParams.Metadata.Name);
+                results = scripter.GetTableScripts(metadata.Name, metadata.Schema);
             }
-            else if (scriptingParams.Operation == ScriptOperation.Create)
+            else if (metadata.MetadataType == MetadataType.SProc)
             {
-                script = string.Format(
-@"CREATE {0}.{1}",
-                scriptingParams.Metadata.Schema, scriptingParams.Metadata.Name);
+                results = scripter.GetStoredProcedureScripts(metadata.Name, metadata.Schema);
             }
-            else if (scriptingParams.Operation == ScriptOperation.Update)
+            else if (metadata.MetadataType == MetadataType.View)
             {
-                script = string.Format(
-@"UPDATE {0}.{1}",
-                scriptingParams.Metadata.Schema, scriptingParams.Metadata.Name);
+                results = scripter.GetViewScripts(metadata.Name, metadata.Schema);
             }
-            else if (scriptingParams.Operation == ScriptOperation.Insert)
-            {
-                script = string.Format(
-@"INSERT {0}.{1}",
-                scriptingParams.Metadata.Schema, scriptingParams.Metadata.Name);
+
+            StringBuilder builder = null;
+            if (results != null) 
+            {                                
+                builder = new StringBuilder();                             
+                foreach (var result in results)
+                {
+                    builder.AppendLine(result);
+                    builder.AppendLine();
+                }
             }
-            else if (scriptingParams.Operation == ScriptOperation.Delete)
+            return builder != null ? builder.ToString() : null;
+        }
+
+        /// <summary>
+        /// Not yet implemented
+        /// </summary>
+        private static string ScriptAsUpdate(
+            IBindingContext bindingContext,
+            ConnectionInfo connInfo,
+            ObjectMetadata metadata)
+        {
+            return null;
```

Comment:
this PR is only focused on the scripting as -> create scenario.  I'll fill in the implementations for the other scripting commands in a subsequent PR.